pyanaconda/ui/tui/spokes/storage.py

```diff
@@ -203,33 +208,36 @@ def refresh(self, args=None):
         #self.selected_disks = self.data.ignoredisk.onlyuse[:]
         self.autopart = self.data.autopart.autopart
 
+        self._container = ListColumnContainer(1, spacing=1)
+
         message = self._update_summary()
 
         # loop through the disks and present them.
-        for disk in self.disks:
+        for i, disk in enumerate(self.disks):
             disk_info = self._format_disk_info(disk)
-            c = CheckboxWidget(title="%i) %s" % (self.disks.index(disk) + 1, disk_info),
-                               completed=(disk.name in self.selected_disks))
-            self._window += [c, ""]
+            c = CheckboxWidget(title=disk_info, completed=(disk.name in self.selected_disks))
+            self._container.add(c, self._update_disk_list_callback, i)
 
         # if we have more than one disk, present an option to just
         # select all disks
         if len(self.disks) > 1:
-            c = CheckboxWidget(title="%i) %s" % (len(self.disks) + 1, _("Select all")),
-                                completed=(self.selection == len(self.disks)))
+            c = CheckboxWidget(title=_("Select all"), completed=(self.selection == len(self.disks)))
+            self._container.add(c, self._select_all_disks_callback, len(self.disks))
 
-            self._window += [c, ""]
+        self.window.add_with_separator(self._container)
+        self.window.add_with_separator(TextWidget(message))
 
-        self._window += [TextWidget(message), ""]
-
-        return True
-
-    def _select_all_disks(self):
+    def _select_all_disks_callback(self, data):
         """ Mark all disks as selected for use in partitioning. """
+        self.selection = data
```

Comment:
`self.selection` can be replaced with something like the `self.selected_all` flag.